### ISSUE TITLE
restify.createJsonClient() wrapper for api

### ISSUE THREAD

**Issue by dreadjr** [restify/node-restify#241]
What is the best way to wrap a restify server api with createJsonClient(), so it would be more natural to use from an client api prespective.

For instance.

then you could wrap all the calls in functions like this.

// constructor
function api() {
  this.client = restify.createJsonClient();
}

function apiCallget(cb) {
  var self = this;
  self.client.get('/apiCall/', function(err, req, res, obj) {
    cb(err, obj);
  });
}

But is there a good way to make it modular so i could load certain "calls" (middleware/plugin) for instance.
Basically how should i use the JsonClient when making a wrapper for an api?

// constructor
function api() {
}

// these could be in other files
api.use(apiCall());  
api.use(otherApiCall());

Any ideas?
